src/app/(sidebar)/transactions-explorer/tx/[tx]/components/TransactionDetails.tsx

```diff
@@ -0,0 +1,199 @@
+import { StrKey } from "@stellar/stellar-sdk";
+import { Asset, Icon, Loader, Text } from "@stellar/design-system";
+import { rpc as StellarRpc } from "@stellar/stellar-sdk";
+import { useEffect, useState, ReactNode, MouseEventHandler } from "react";
+import { useRouter } from "next/navigation";
+import { parseNumberAndBigInt, parse as JsonParse } from "lossless-json";
+
+import { JsonCodeWrapToggle } from "@/components/JsonCodeWrapToggle";
+import { Box } from "@/components/layout/Box";
+import { PageCard } from "@/components/layout/PageCard";
+import { PrettyJsonTransaction } from "@/components/PrettyJsonTransaction";
+import { SdsLink } from "@/components/SdsLink";
+
+import { formatTimestamp } from "@/helpers/formatTimestamp";
+import * as StellarXdr from "@/helpers/StellarXdr";
+import { delayedAction } from "@/helpers/delayedAction";
+import { buildEndpointHref } from "@/helpers/buildEndpointHref";
+
+import { Routes } from "@/constants/routes";
+import { useStore } from "@/store/useStore";
+
+const InfoField = ({ label, value }: { label: string; value: ReactNode }) => (
+  <Box gap="xs" direction="row" align="center" addlClassName="InfoFieldItem">
+    <div className="InfoFieldItem__label">{label}</div>
+    <div className="InfoFieldItem__value">{value ?? "-"}</div>
+  </Box>
+);
+
+export function TransactionDetails({
+  tx,
+}: {
+  tx:
+    | StellarRpc.Api.GetSuccessfulTransactionResponse
+    | StellarRpc.Api.GetFailedTransactionResponse;
+}) {
+  const { endpoints } = useStore();
+  const router = useRouter();
+  const [isCodeWrapped, setIsCodeWrapped] = useState(false);
+  const [xdrJson, setXdrJson] = useState<Record<string, unknown> | null>(null);
+  const [xdr, setXdr] = useState("");
+  const innerTx = tx.feeBump
+    ? // @ts-expect-error fee bump tx has innerTx.
+      tx.envelopeXdr.value().tx().innerTx().value().tx()
+    : tx.envelopeXdr.value().tx();
+  const sourceAccount = StrKey.encodeEd25519PublicKey(
+    innerTx.sourceAccount().value(),
+  );
+
+  const goToAccount: MouseEventHandler = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    delayedAction({
+      action: () => {
+        endpoints.updateParams({ account_id: sourceAccount });
+        router.push(Routes.ENDPOINTS_ACCOUNTS_SINGLE);
+      },
+      delay: 100,
+    });
+  };
+
+  const goToTransaction: MouseEventHandler = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    delayedAction({
+      action: () => {
+        endpoints.updateParams({ transaction: tx.txHash });
+        router.push(Routes.ENDPOINTS_TRANSACTIONS_SINGLE);
+      },
+      delay: 100,
+    });
+  };
+
+  useEffect(() => {
+    const parse = async () => {
+      await StellarXdr.initialize();
```

Comment:
There's a `useIsXdrInit()` hook ([example](https://github.com/stellar/laboratory/blob/main/src/app/(sidebar)/xdr/to/page.tsx#L20)) we can use here. We should probably put this outside of the effect to check only when the component mounts.